### PR TITLE
Add custom catalog source

### PR DESCRIPTION
**Pull request by kurtismullins**
This Pull Request adds a custom catalog which can be used for testing purposes.

### CODE REVIEW

**Review by alecmerdler — CHANGES_REQUESTED** [review]
```shell
$ docker build -t quay.io/alecmerdler/quay-operator-catalog .
Sending build context to Docker daemon  914.4kB
Step 1/10 : FROM quay.io/operator-framework/upstream-registry-builder as builder
 ---> 015c5c095bba
Step 2/10 : COPY manifests manifests
 ---> 8784000a0eb0
Step 3/10 : RUN ./bin/initializer -o ./bundles.db
 ---> Running in 5a0f9485f5e8
time="2020-04-18T21:37:48Z" level=info msg="loading Bundles" dir=manifests
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=manifests load=bundles
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=quay-operator load=bundles
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=1.1.0 load=bundles
time="2020-04-18T21:37:48Z" level=info msg="found csv, loading bundle" dir=manifests file=quay.v1.0.2.clusterserviceversion.yaml load=bundles
time="2020-04-18T21:37:48Z" level=info msg="loading bundle file" dir=manifests/quay-operator/1.1.0 file=package.yaml load=bundle
time="2020-04-18T21:37:48Z" level=info msg="loading bundle file" dir=manifests/quay-operator/1.1.0 file=quay.v1.0.2.clusterserviceversion.yaml load=bundle
time="2020-04-18T21:37:48Z" level=info msg="loading bundle file" dir=manifests/quay-operator/1.1.0 file=redhatcop.redhat.io_quayecosystems_crd.yaml load=bundle
time="2020-04-18T21:37:48Z" level=info msg="loading Packages and Entries" dir=manifests
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=manifests load=package
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=quay-operator load=package
time="2020-04-18T21:37:48Z" level=info msg=directory dir=manifests file=1.1.0 load=package
time="2020-04-18T21:37:48Z" level=fatal msg="permissive mode disabled" error="error loading manifests from directory: could not decode contents of file manifests/quay-operator/1.1.0/package.yaml into package: error converting YAML to JSON: yaml: line 3: did not find expected '-' indicator"
The command '/bin/sh -c ./bin/initializer -o ./bundles.db' returned a non-zero code: 1
```

File: deploy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM quay.io/operator-framework/upstream-registry-builder as builder
```
Comment (alecmerdler) [inline deploy/Dockerfile:1]: Be sure to create a `.dockerignore` file and add `deploy/` to it (so the Operator container SHA is unaffected by changes to this directory).

**Review by sabre1041 — CHANGES_REQUESTED** [review]
The CSV filename should be `quay.v1.1.0....`

Recommend using the `operator-sdk` to generate the csv as it will automatically build the ALM Examples.


The following is a command that I had used previously

```

operator-sdk generate csv --from-version=1.0.2 --csv-version=1.1.0 --update-crds --operator-name=quay
```

And then remove the OCP 3.x CRD that comes along for the ride


In addition, why are we not populating the CSV and updating the `package.yml` in the `deploy/olm-catalog` folder for the manifest with existing assets?

**Comment by kurtismullins** [comment]
> In addition, why are we not populating the CSV and updating the `package.yml` in the `deploy/olm-catalog` folder for the manifest with existing assets?

Thanks @sabre1041! The reason I'm not doing that in _this_ commit is that this is simply used to create a catalog that QE can test from. We can do whatever finalization is needed for the release in a separate Pull Request

**Comment by kurtismullins** [comment]
I'm going to go ahead and merge this in for QE's sake. There will probably be additional changes in an upcoming PR.

**Pull request merged**